heartpoints.sh

```diff
@@ -610,23 +692,58 @@ gcloud_configure() {
     gcloud_cli container clusters get-credentials heartpoints-org --zone us-central1-a --project heartpoints-org
 }
 
-# Misc functions
+brew_app_dir_path() {
+    createAndReturnPath "$(devEnvironmentPath)/brewAppDir"
+}
 
-brew_cask_caskIsInstalled() { local caskName=$1
-    brew cask list | grep "${caskName}" > /dev/null 2>&1
+devEnvironmentPath() {
+    createAndReturnPath "./devEnvironment"
 }
 
-brew_cask_installCask() { local caskName=$1
-    brew_cask_installCaskroom
-    if command_does_not_exist "${caskName}"; then
-        brew cask install "${caskName}"
+hp_brew() { local args="$@"
+    if command_does_not_exist "$(homebrew_cli_path)"; then
+        brew_install_brew_itself
     fi
+    $(homebrew_cli_path) "$@"
 }
 
-brew_cask_installCaskroom() {
-    if ! brew info cask &>/dev/null; then
-        brew tap caskroom/cask
-    fi
+homebrew_cli_path() {
+    homebrew_bin_path brew
+}
+
+homebrew_bin_path() { local caskOrPackageName=$1
+    echo "$(homebrew_install_dir)/bin/${caskOrPackageName}"
+}
+
+homebrew_install_dir() {
+    createAndReturnPath "$(devEnvironmentPath)/homebrew"
+}
+
+homebrew_tar_path() {
+    createAndReturnPath "$(devEnvironmentPath)/homebrewTar"
+}
+
+homebrew_tar_download_log_path() {
+    hp_log_path "homebrew_tar_download.log"
+}
+
+brew_install_log_path() {
+    hp_log_path "brew_install_brew_itself.log"
+}
+
+brew_install_brew_itself() {
```

Comment:
perhaps we should be explicit about which functions are installed locally within this repo vs globally

Reply:
can you elaborate on this? :)

Reply:
some of the brew installations in this PR  install into a directory within this repository. and others, like Virtualbox, are installed globally for that user/computer. It would be great to have a comment or convention to differentiate this